### PR TITLE
include new feature calculator for dickey fuller p-value

### PR DESCRIPTION
**Pull request by mynameisvinn**
@jneuff in reference to https://github.com/blue-yonder/tsfresh/issues/5

returning p-value from dickey fuller calculation

### CODE REVIEW

**Comment by MaxBenChrist** [comment]
Nice, but we should wrap the call to the ad fuller test to not calculate I several times. We can use a decorator for that. See http://stackoverflow.com/questions/815110/is-there-a-decorator-to-simply-cache-function-return-values

File: tsfresh/feature_extraction/feature_calculators.py [hunk 1]
```diff
@@ -276,6 +276,27 @@ def augmented_dickey_fuller(x):
...
+    """
+    The Augmented Dickey-Fuller is a hypothesis test which checks whether a unit root is present in a time
+    series sample. This feature calculator returns the value of the corresponding p-value.
+
+    :param x: the time series to calculate the feature of
+    :type x: pandas.Series
+    :return: the value of this feature
+    :return type: float
+    """
+    maxlag = 1
```
Comment (MaxBenChrist) [inline tsfresh/feature_extraction/feature_calculators.py]: Line 290 to 295 should be moved into its own method and wrapped with a cache decorator

File: tsfresh/feature_extraction/feature_calculators.py [hunk 2]
```diff
@@ -251,6 +251,26 @@ def mean_autocorrelation(x):
         return np.nanmean(r / var)
 
 
+def memoize(f):
+    cache = {}
```
Comment (MaxBenChrist) [inline tsfresh/feature_extraction/feature_calculators.py]: I dont like that the cache is not size limited. have you looked at some caching libraries?

https://pythonhosted.org/cachetools/

Comment (mynameisvinn) [inline tsfresh/feature_extraction/feature_calculators.py]: makes sense. i have not but will take a look.

**Comment by mynameisvinn** [comment]
in all sorts of pain right now - cant figure out why caching decorator is failing tests... will review next weekend

**Comment by mynameisvinn** [comment]
using decorator to cache results from augmented dickey fuller calculation for (1) test statistic and (2) p-value. i think this is what u had in mind but not 100% certain...

latest pr is failing all sorts of tests but not anything related to dickey fuller calculations.

**Comment by MaxBenChrist** [comment]
the failing unit tests are related to the caching mechanism. do you want to look into that?

**Comment by MaxBenChrist** [comment]
@mynameisvinn are you still interested in finishing this or should we take over?

**Comment by mynameisvinn** [comment]
@MaxBenChrist sorry for slow response - busy year end work stuff. i wont have much free time until the new year, so go ahead without me!

see u in the new year!

**Comment by MaxBenChrist** [comment]
thanks for the response, see you!

**Pull request closed without merging**